### PR TITLE
Intro

### PR DESCRIPTION
**Pull request by ghost**
Basic introduction and guide on wot++

### CODE REVIEW

File: docs/introduction.md [hunk 1]
```diff
@@ -1,2 +1,50 @@
...
+### Raw Strings
+
+Raw strings are as simple as they come what gets entered is what gets stored as a string. The two ways to call it are either `""` which implicitly treats the string as a raw string but if you wish to be explicit you can use. `r#" #"`
+
+Input:
+
+```wot++
+r#"
+    Hello world.
+#"
```
Comment (qookei) [inline docs/introduction.md:17]: You can use any non whitespace character in place of the `#`.

Comment (qookei) [reply docs/introduction.md:17]: Also the format is `r#" ... "#` not `r#" ... #"`

File: docs/introduction.md [hunk 2]
```diff
@@ -1,2 +1,50 @@
...
+### Paragraph Strings
+
+Now what a paragraph string does is removes leading and trailing whitespace. It's syntax is very similar to the raw string it is. `p#" #"`
+
+Input:
+
+```wot++
+p#"
+    Hello world.
+p#"
```
Comment (qookei) [inline docs/introduction.md:33]: Same here

File: docs/introduction.md [hunk 3]
```diff
@@ -1,2 +1,50 @@
...
+### Code Strings
+
+This is for when you want to store a string that is code It will remove common leading white space. Like the last two it follows as such. `c#" #"`
+
+Input:
+
+```wot++
+    c#"
+        Hello World.
+    #"
```
Comment (qookei) [inline docs/introduction.md:49]: Same here

**Review by qookei — CHANGES_REQUESTED** [review]
Personally, I think the writing style is too informal for docs, but I'm not sure what others think.

**Comment by Jackojc** [comment]
So, just a few things to address. 

First off, I think this would work better as a reference rather than an introductory guide for beginners. I feel like a beginner's guide should present an example project and then walk you through the basics of the language to get from nothing to a small working project. This would let you simultaneously show off some of the more basic features while also showing the intended use-case for wot++. 

As it currently stands, this seems more like a list of features that _you personally_ think are cool but not something super relevant to a beginner. For example, It's probably not useful for a beginner to know about every type of string there is for writing "Hello, World!".

The style of writing also feels more personal and informal than I think documentation should be. Some examples of what I mean include:
- > Strings can be identified by a `""` or a `''` which is pretty nice.
- > Strings are the data we focus on in this language
- > Raw strings are as simple as they come
- > Comments are always useful so we included them for you

etc.

Aside from these more specific examples there are more general grammatical errors including capitalisation and run on sentences.

To get a feel for the general direction we want to take this introduction guide, it might be useful to look at examples from other languages as a rough guide:
<https://docs.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/tutorials/numbers-in-csharp>
<https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/introduction>
<https://bodil.lol/parser-combinators/>

Obviously these examples are pretty big and complicated but the general idea is still there.

I'm going to close this PR for now.

**Comment by jlagarespo** [comment]
Perhaps we could use the [github wiki feature](https://docs.github.com/en/github/building-a-strong-community/about-wikis) for writing documentation, tutorials, etc.

**Pull request closed without merging**